lib/Transforms/Utils/SimplifyCFG.cpp

```diff
@@ -138,12 +139,12 @@ class SimplifyCFGOpt {
   bool SimplifySwitch(SwitchInst *SI, IRBuilder<> &Builder);
   bool SimplifyIndirectBr(IndirectBrInst *IBI);
   bool SimplifyUncondBranch(BranchInst *BI, IRBuilder <> &Builder);
-  bool SimplifyCondBranch(BranchInst *BI, IRBuilder <>&Builder);
+  bool SimplifyCondBranch(BranchInst *BI, IRBuilder <>&Builder, bool AllowFoldCondBranchOnPHI); // HLSL Change
 
 public:
   SimplifyCFGOpt(const TargetTransformInfo &TTI, const DataLayout &DL,
-                 unsigned BonusInstThreshold, AssumptionCache *AC)
-      : TTI(TTI), DL(DL), BonusInstThreshold(BonusInstThreshold), AC(AC) {}
+                 unsigned BonusInstThreshold, AssumptionCache *AC, /*HLSL Change */bool AllowFoldCondBranchOnPHI)
+      : TTI(TTI), DL(DL), BonusInstThreshold(BonusInstThreshold), AC(AC), /*HLSL Change*/ AllowFoldCondBranchOnPHI(AllowFoldCondBranchOnPHI) {}
```

Comment:
In case the clang error is confusing, it just wants the initializers here to be in the same order as they are declared. So swapping AC and AllowFoldCondBranchOnPHI will solve it.

Reply:
Thanks!